src/script/lua_api/l_mapgen.cpp

```diff
@@ -986,11 +1026,26 @@ int ModApiMapgen::l_set_gen_notify(lua_State *L)
 		}
 	}
 
+	if (lua_istable(L, 3)) {
+		lua_pushnil(L);
+		while (lua_next(L, 3)) {
+			emerge->gen_notify_on_custom.insert(readParam<std::string>(L, -1));
```

Comment:
`emerge->gen_notify_on_custom` could be pre-allocated using `lua_objlen` or similar.

Reply:
`lua_next` is actually like pairs and not ipairs, so `lua_objlen` is not necessarily correct.
reading an array to a vector is a common pattern and IMO some later PR should be cleaning that up.